main.py

```diff
@@ -45,75 +45,84 @@
-            version="4.2",
-            edition="enterprise",
-            cluster=False,
-            suite="4.2",
-            scheme="neo4j",
-            download=teamcity.DockerImage(
-                "neo4j-enterprise-4.2.12-docker-loadable.tar"),
-            stress_test_duration=0))
-
-        configurations.append(neo4j.Config(
-            name="4.3-tc-enterprise",
-            image="neo4j:4.3.4-enterprise",
-            version="4.3",
-            edition="enterprise",
-            cluster=False,
-            suite="4.3",
-            scheme="neo4j",
+    def generate_config(version, enterprise, cluster, scheme, stress_test):
+        assert (cluster and scheme == "neo4j"
+                or not cluster and scheme in ("neo4j", "bolt"))
+        edition = "enterprise" if enterprise else "community"
+        name = "%s-%s%s-%s" % (version, edition,
+                               "-cluster" if cluster else "", scheme)
+        return neo4j.Config(
+            name=name,
+            image="neo4j:%s%s" % (version, "-enterprise" if enterprise else ""),
+            version=version,
+            edition=edition,
+            cluster=cluster,
+            suite=version,
+            scheme=scheme,
+            download=None,
+            stress_test_duration=stress_test
+        )
+
+    def generate_tc_config(version, enterprise, cluster, scheme, stress_test):
+        if not in_teamcity:
+            return None
+        assert (cluster and scheme == "neo4j"
+                or not cluster and scheme in ("neo4j", "bolt"))
+        edition = "enterprise" if enterprise else "community"
+        name = "%s-tc-%s%s-%s" % (version, edition,
+                                  "-cluster" if cluster else "", scheme)
+        version_without_drop = ".".join(version.split(".")[:2])
+        return neo4j.Config(
+            name=name,
+            image="neo4j:%s%s" % (version, "-enterprise" if enterprise else ""),
+            version=version_without_drop,
+            edition=edition,
+            cluster=cluster,
+            suite=version_without_drop,
+            scheme=scheme,
             download=teamcity.DockerImage(
-                "neo4j-enterprise-4.3.4-docker-loadable.tar"),
-            stress_test_duration=0))
+                "neo4j-%s-%s-docker-loadable.tar" % (edition, version)
+            ),
+            stress_test_duration=stress_test
+        )
+
+    configurations = [
+        generate_config(version_, enterprise_, cluster_, scheme_, stress_test_)
+        for (version_, enterprise_, cluster_, scheme_, stress_test_) in (
+            # LTS version
+            # 3.5 servers only support routing scheme if configured as cluster
+            ("3.5",    False,       False,    "bolt",   0),
+            ("3.5",    True,        False,    "bolt",   0),
+            ("3.5",    True,        True,     "neo4j", 60),
+            # not officially supported versions
+            ("4.0",    True,        False,    "neo4j",  0),
```

Comment:
4.0 has reached its end of life on July 14 (sorry I did not catch this earlier)
Shall we get rid of 4.0 altogether?

Reply:
I'd like to get the opinion of the other team members on this. I feel like it would be ok to drop it.

Reply:
I would be fine with it, I think the differences in Cypher and Indizes are something that mostly effects things like OGM. 
3.5 we need to keep anyway, right?

Reply:
Yep, 3.5 EOL is November 28, 2021